test/unit/versioned_libs/versioned_libs_analysis_test.bzl

```diff
@@ -0,0 +1,235 @@
+        target_compatible_with = ["@platforms//os:linux"],
+    )
+
+    rust_shared_library(
+        name = "linux_suffix_version",
+        srcs = ["a.rs"],
+        deps = [":import_libcheryl.so.3.8", ":import_libcheryl.so"],
+        target_compatible_with = ["@platforms//os:linux"],
+    )
+    cc_import(
+        name = "import_libcheryl.so.3.8",
+        shared_library = "libcheryl.so.3.8",
+    )
+    cc_binary(
+        name = "libcheryl.so.3.8",
+        srcs = ["b.c"],
+        linkshared = True,
+    )
+    cc_import(
+        name = "import_libcheryl.so",
+        shared_library = "libcheryl.so",
+    )
+    copy_file(
+        name = "copy_unversioned",
+        src = ":libcheryl.so.3.8",
+        out = "libcheryl.so",
+    )
+    suffix_version_test(
+        name = "linux_suffix_version_test",
+        target_under_test = ":linux_suffix_version",
+        target_compatible_with = ["@platforms//os:linux"],
+    )
+
+    return [
+        ":linux_no_version_test",
+        ":linux_suffix_version_test",
+    ]
+
+def _test_macos():
+    rust_shared_library(
+        name = "no_version",
+        srcs = ["a.rs"],
+        deps = [":import_liblana.dylib"],
+        target_compatible_with = ["@platforms//os:macos"],
+    )
+    cc_import(
+        name = "import_liblana.dylib",
+        shared_library = "liblana.dylib",
+    )
+    cc_binary(
+        name = "liblana.dylib",
+        srcs = ["b.c"],
+        linkshared = True,
+    )
+    no_version_test(
+        name = "macos_no_version_test",
+        target_under_test = ":no_version",
+        target_compatible_with = ["@platforms//os:macos"],
+    )
+
+    rust_shared_library(
+        name = "prefix_version",
+        srcs = ["a.rs"],
+        deps = [":import_libpam3.8.dylib"],
+        target_compatible_with = ["@platforms//os:macos"],
+    )
+    cc_import(
+        name = "import_libpam3.8.dylib",
+        shared_library = "libpam3.8.dylib",
+    )
+    cc_binary(
+        name = "libpam3.8.dylib",
+        srcs = ["b.c"],
+        linkshared = True,
+    )
+    prefix_version_test(
+        name = "macos_prefix_version_test",
+        target_under_test = ":prefix_version",
+        target_compatible_with = ["@platforms//os:macos"],
+    )
+
+    return [
+        ":macos_no_version_test",
+        ":macos_prefix_version_test",
+    ]
+
+def _test_windows():
+    rust_shared_library(
+        name = "windows_no_version",
+        srcs = ["a.rs"],
+        deps = [":import_malory.dll"],
+        target_compatible_with = ["@platforms//os:windows"],
+    )
+    cc_import(
+        name = "import_malory.dll",
+        interface_library = ":malory.lib",
+        shared_library = "malory.dll",
+    )
+    cc_binary(
+        name = "malory.dll",
+        srcs = ["b.c"],
+        linkshared = True,
+    )
+    native.filegroup(
+        name = "malory_interface_lib",
+        srcs = [":malory.dll"],
+        output_group = "interface_library",
+    )
+    native.genrule(
```

Comment:
Small nit: could `copy_file` be used here in this function? Is there a difference in it's behavior on windows than on linux?

Reply:
Ah yes I believe it should work just the same. I was switching between `genrule` and `copy_file` while trying to debug a naming issue with the interface library (I think `cc_library` generates the lib file as `something.dll.lib` that Windows doesn't recognize since it expects `something.lib` to live next to `something.dll`, hence the renaming).

Reply:
Ah, ok, Sounds good. Could you move the `genrule` to `copy_file` for consistency if it works?

Reply:
Yep I just did. Let's see if CI is happy with it :)

Reply:
Oh it doesn't and now I remember why it is a `genrule`. Looks like the `interface_library` `filegroup` generates multiple outputs (even though by the time the `genrule` runs `$<` is a single input 🤔) and `copy_file` chokes on it.

Reply:
That's fine. Maybe just add a comment about if you've still got the code open? 🙏

Reply:
For sure, doing now.

Reply:
ok I actually had a typo and `copy_file` works fine, sorry for the confusion 😄